lib/SPIRV/OCL20ToSPIRV.cpp

```diff
@@ -509,7 +509,7 @@ OCL20ToSPIRV::visitCallNDRange(CallInst *CI,
     default:
       assert(0 && "Invalid number of arguments");
     }
-    return getSPIRVFuncName(OpBuildNDRange);
+    return getSPIRVFuncName(OpBuildNDRange, "_" + std::to_string(Len) + "D");

```

Comment:
The problem is in ndrange_2D and ndrange_3D. They are mangled the same way but have different argument types (2D and 3D arrays).